### ISSUE TITLE
unique=True validation not working with models.CharField with choices

### ISSUE THREAD

**Issue by Ekluv** [tomchristie/django-rest-framework#5004]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce
Lets take a model 
``` python
MY_CHOICES = (
    ('choice1', _('choice 1')),
    ('choice2', _('choice 1')),
)

class Poll(models.Model):
    form_name = models.CharField('name', max_length=254,
                                 unique=True, choices=MY_CHOICES)

class PollSerializer(serializers.ModelSerializer):
	class Meta:
		model = Poll
		fields = '__all__'

class PollCreateView(generics.CreateAPIView):
	serializer_class = PollSerializer
```
Then making a POST request with `{"form_name": "choice1"}` first time returns `{
  "id": 1,
  "form_name": "choice1"
}`

then by making same request again with data  `{"form_name": "choice1"}` instead of giving valid error response, django's IntegrityError exception is thrown.

This behaviour is only coming if models.CharField is used with choices and unique=True and not when models.CharField is used without choices

## Expected behavior
Valid error response should be returned instead of throwing django's IntegrityError exception
like `{
  "form_name": [
    "poll with this name already exists."
  ]
}`

## Actual behavior

Throwing django's IntegrityError exception when models.CharField is used with choices and unique=True

**Comment by Ekluv** [comment]
any update on this ?

**Comment by tomchristie** [comment]
Confirmed...

```python
>>> from rest_framework.serializers import ModelSerializer
>>> from snippets.models import Poll
>>> class Foo(ModelSerializer):
...     class Meta:
...         model = Poll
...         fields = '__all__'
... 
>>> print(Foo())
Foo():
    id = IntegerField(label='ID', read_only=True)
    form_name = ChoiceField(choices=(('choice1', 'choice 1'), ('choice2', 'choice 1')), label='Name')
```

**Comment by jpadilla** [comment]
Seems like [this](https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/utils/field_mapping.py#L125) is the culprit.

**Comment by tomchristie** [comment]
Indeed yup, some careful rejigging will be needed there, since there are *some* keyword arguments that we don't want to include for `ChoiceField`.
Right now we're erroneously stripping `validators` tho, so we want to add *just* that back in.

**Comment by jpadilla** [comment]
@tomchristie just removing that early return seems to work since all the checks below have constraint around the field type already.

```
PollSerializer():
    id = IntegerField(label='ID', read_only=True)
    form_name = ChoiceField(choices=((u'choice1', u'choice 1'), (u'choice2', u'choice 1')), label='Name', validators=[<UniqueValidator(queryset=Poll.objects.all())>])
```

**Comment by Ekluv** [comment]
@jpadilla: in #5026 just removing early return may create issues. It will also include validation for `max_value` and `min_value` which may not be required for choice field.
#5028 take cares of this.
@tomchristie : please give your concerns if any.

**Comment by jpadilla** [comment]
Fixed via #5028
